contracts/external/dao-migrator/src/testing/test_migration.rs

```diff
@@ -0,0 +1,147 @@
+use std::borrow::BorrowMut;
+
+use dao_core::query::SubDao;
+
+use crate::{
+    testing::{helpers::ExecuteParams, state_helpers::query_state_v2},
+    ContractError,
+};
+
+use super::{
+    helpers::VotingType,
+    setup::{execute_migration, setup_dao_v1},
+    state_helpers::query_state_v1,
+};
+
+#[test]
+fn test_execute_migration() {
+    let (mut app, module_addrs, v1_code_ids) = setup_dao_v1(VotingType::Cw20);
```

Comment:
is there a version of this test with `VotingType::Cw4`? that would be nice to have. great work on that instantiation abstraction btw.